### DISCUSSION TITLE
SessionProvider in Root Layout (Next.js App Router)

### DISCUSSION THREAD

**Discussion by howard36** [nextauthjs/next-auth#8137] [Help]
How do I include `SessionProvider` in my root layout?

Here is what I'm currently trying:

`app/layout.tsx`

```jsx
import Providers from './providers'

export default function RootLayout({
  children,
}: {
  children: React.ReactNode
}) {
  return (
    <html lang="en">
      <body>
        <Providers>{children}</Providers>
      </body>
    </html>
  )
}
```

`app/providers.tsx`
```jsx
'use client';

import { SessionProvider } from 'next-auth/react';

export default function Providers({
  children,
}: {
  children: React.ReactNode;
}) {
  return <SessionProvider>{children}</SessionProvider>;
}
```

But this method doesn't pass the `session` prop to `SessionProvider` (and unlike `_app.tsx` from the Pages Router, I can't get `session` from the props to a Layout, since the only prop it accepts is `children`)

---

## Version Info
- Next.js 13.4.10
- Next-Auth 4.22.3

## Similar Discussions (didn't help much unfortunately)
- https://github.com/nextauthjs/next-auth/issues/7760
- https://github.com/nextauthjs/next-auth/discussions/5956

**Comment by moinerus** [comment]
Would love t see an example of this!
